azkaban-common/src/test/java/azkaban/executor/InteractiveTestJob.java

```diff
@@ -137,6 +138,12 @@ public void succeedJob(final Props generatedProperties) {
     }
   }
 
+  public void ignoreCancel() {
+    synchronized (this) {
+      this.ignoreCancel = true;
```

Comment:
Maybe, but I was following the pattern used in this class. Let's have it like this and refactor all those flags to volatile in a later commit?